### PR TITLE
Add a nice debouncer from WPILib

### PR DESCRIPTION
**Pull request by mikemag**
Copied the debouncer from current WPILib and changed from their JNI call to get time to the standard Java `System.nanoTime()`. I placed it in a similar package as is in WPILib since I felt like they had made a good choice.

### CODE REVIEW

**Comment by silverlightning926** [comment]
Is `1e-9` the conversion constant from nanoseconds to seconds? Found it as it was currently used here ` m_prevTimeSeconds = System.nanoTime() * 1e-9;` and a few other places. I actually didn't know Java supports that kind of notation for exponents, so I had to go search the notation up, originally I thought that `e` was refrencing Euler's number / the "e" constant. Maybe that can be cleared up using a comment or using `Math.pow(base, exponent)` instead? Also currently seems like a magic constant just floating there. I don't know if we want a constants file to keep all the constants in one place. There is also a `TimeUnit.SECONDS.convert(duration, sourceUnit)` function that can be used instead.

**Comment by silverlightning926** [comment]
Also doesn't look like this merge causes any merge causes any merge conflicts, so that's nice! But I was just wondering what the plan would be after the pr? Would we then need to merge the `main` branch with each team branch so that they can access an use the updated StealthLib?

**Comment by mikemag** [comment]
> Is `1e-9` the conversion constant from nanoseconds to seconds? 

Yes! I'll add a comment. Scientific notation with this syntax is supported in all programming languages that I know of, and you'll notice it even as the default formatting of quite large or small doubles with `System.out.println`.

`Math.pow` will disappoint you... it will compute the power every time you call it. You'd like to hope they'd bake the constant in, but nope!

The `TimeUnit` class is... interesting. Notice that all of the conversion functions take and produce longs. Strange choice, and not useful for dealing with sub-second duration, as we are here.

**Comment by mikemag** [comment]
> Also doesn't look like this merge causes any merge causes any merge conflicts, so that's nice! But I was just wondering what the plan would be after the pr? Would we then need to merge the `main` branch with each team branch so that they can access an use the updated StealthLib?

I haven't been pushing main down to the team branches. Instead, I've been having them cherry pick little things like this into their branches. It helps avoid any merge conflicts with their stuff, and can be done pretty much regardless of the current state of their branch. Both work, though, and if we wanna try to push all of main down we can, we'll just have to coordinate with them a little more.

**Comment by silverlightning926** [comment]
Yea, we would have to wait till everyone has pushed, merge everyone's changes into main and then push main to everyone. Maybe that would be possible to do after Sunday's (11/20/2022) meeting? That way we can also push the actions out to the team branches. We would need everyone to commit and push what they have at that meeting though.

**Comment by silverlightning926** [comment]
Otherwise everything looks good to me, going to accept and complete the merge!

File: StealthLib/src/main/java/org/stealthrobotics/library/math/filter/Debouncer.java
Comment (silverlightning926) [inline StealthLib/src/main/java/org/stealthrobotics/library/math/filter/Debouncer.java:64]: Perfect! This comment really help!

**Review by silverlightning926 — APPROVED** [review]
Everything looks good to me, going to approve and complete the merge!

**Pull request merged**
